### PR TITLE
Replace blacklisted cache entry log message with metric [BA-6311]

### PR DESCRIPTION
**Pull request by cjllanwarne**
What do you mean "metricable" isn't a word?

\<looks quickly on google\>

...

"I am **not** using [Metrizable](https://www.yourdictionary.com/metrizable)...!"

---

Additional commentary and/or things worth sanity checking me on:

* We lose "root workflow ID" and "bucket" from the log message.
* We gain task name and hog group in the metric path.

### CODE REVIEW

File: engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/job/EngineJobExecutionActor.scala [hunk 1]
```diff
@@ -701,8 +707,20 @@ class EngineJobExecutionActor(replyTo: ActorRef,
     }
   }
 
-  private def invalidateCacheHitAndTransition(ejeaCacheHit: EJEACacheHit, data: ResponsePendingData, reason: Throwable) = {
-    logCacheHitFailure(data, reason)
+  private def metricizeCacheHitFailure(data: ResponsePendingData, failureCategory: String): Unit = {
+    val workflowsToDeleteMetadataMetricPath: NonEmptyList[String] =
```
[the remark this answers is not in the document]
Comment (mcovarr) [reply engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/job/EngineJobExecutionActor.scala:711]: needs fixin 🙂

File: server/src/test/scala/cromwell/engine/workflow/lifecycle/execution/ejea/EngineJobExecutionActorSpecUtil.scala [hunk 2]
```diff
@@ -118,6 +118,12 @@ private[ejea] trait HasJobFailureResponses { self: EngineJobExecutionActorSpec =
...
     new Exception("Deliberate failure for test case: failed to copy cache outputs!") with NoStackTrace
+
   // Need to delay making the response because job descriptors come from the per-test "helper", which is null outside tests!
-  def failedToCopyResponse(attemptNumber: Int) = CopyingOutputsFailedResponse(helper.jobDescriptorKey, attemptNumber, copyFailureReason)
+  def loggableFailedToCopyResponse(attemptNumber: Int) = CopyingOutputsFailedResponse(helper.jobDescriptorKey, attemptNumber, LoggableCacheCopyError(copyFailureReason))
+  def metricableFailedToCopyResponse(attemptNumber: Int) = CopyingOutputsFailedResponse(helper.jobDescriptorKey, attemptNumber, MetricableCacheCopyError(HasCopyFailureResponses.metricContext))
+}
+
+object HasCopyFailureResponses {
+  val metricContext = "metriccontext"
```
Comment (mcovarr) [inline server/src/test/scala/cromwell/engine/workflow/lifecycle/execution/ejea/EngineJobExecutionActorSpecUtil.scala:128]: is this the intended value?

Comment (cjllanwarne) [reply server/src/test/scala/cromwell/engine/workflow/lifecycle/execution/ejea/EngineJobExecutionActorSpecUtil.scala:128]: it's just a test, so I didn't put _too_ much thought into it's value

File: backend/src/main/scala/cromwell/backend/standard/callcaching/StandardCacheHitCopyingActor.scala [hunk 3]
```diff
@@ -254,15 +249,17 @@ abstract class StandardCacheHitCopyingActor(val standardParams: StandardCacheHit
...
   }
 
-  def failAndStop(failure: Throwable) = {
+  def failAndStop(failure: LoggableOrMetricableCacheCopyError): State = {
     context.parent ! CopyingOutputsFailedResponse(jobDescriptor.key, standardParams.cacheCopyAttempt, failure)
     context stop self
     stay()
   }
 
+  def failAndStop(failure: Throwable): State = failAndStop(LoggableCacheCopyError(failure))
```
[the remark this answers is not in the document]
Comment (cjllanwarne) [reply backend/src/main/scala/cromwell/backend/standard/callcaching/StandardCacheHitCopyingActor.scala]: Removed this "make it easy to default to loggable"  method, and added comments where loggable failures are generated.

File: server/src/test/scala/cromwell/engine/workflow/lifecycle/execution/ejea/EngineJobExecutionActorSpecUtil.scala (hunk 2 above)
Comment (cjllanwarne) [reply server/src/test/scala/cromwell/engine/workflow/lifecycle/execution/ejea/EngineJobExecutionActorSpecUtil.scala:128]: Alternative suggestions welcome, of course...

File: backend/src/main/scala/cromwell/backend/BackendCacheHitCopyingActor.scala [hunk 4]
```diff
@@ -1,10 +1,23 @@
...
+  final case class MetricableCacheCopyError(failureCategory: MetricableCacheCopyErrorCategory) extends CacheCopyError
 }
+
+object MetricableCacheCopyErrorCategory {
+  sealed trait MetricableCacheCopyErrorCategory {
+    // The 'stripSuffix is necessary to handle scala 'case object's which get '$' added to the class name during java translation.
+    override def toString: String = getClass.getSimpleName.stripSuffix("$").toLowerCase
+  }
+  final case object BucketBlacklisted extends MetricableCacheCopyErrorCategory
+}
```
Comment (cjllanwarne) [inline backend/src/main/scala/cromwell/backend/BackendCacheHitCopyingActor.scala:23]: newline

File: src/ci/bin/test.inc.sh [hunk 5]
```diff
@@ -475,7 +482,7 @@ cromwell::private::create_centaur_variables() {
             CROMWELL_BUILD_CENTAUR_TEST_DIRECTORY="${CROMWELL_BUILD_CENTAUR_RESOURCES}/${CROMWELL_BUILD_CENTAUR_TYPE}TestCases"
             if test "${CROMWELL_BUILD_BACKEND_TYPE}" = "papi_v2alpha1" || test "${CROMWELL_BUILD_BACKEND_TYPE}" = "papi_v2beta"
             then
-              CROMWELL_BUILD_CENTAUR_CONFIG="${CROMWELL_BUILD_RESOURCES_DIRECTORY}/papi_v2_centaur_application.conf"
```
Comment (mcovarr) [inline src/ci/bin/test.inc.sh:478]: what happened to this file?

File: src/ci/docker-compose/docker-compose-horicromtal.yml [hunk 6]
```diff
@@ -3,6 +3,10 @@
 version: '2.3'
 services:
   # Runs the database initialization but is NOT a workflow-running backend.
+  # Opts out of:
+  # - Running the summarizer
+  # - Running the carboniter
+  # - Running the archived-metadata-deleter
```
Comment (mcovarr) [inline src/ci/docker-compose/docker-compose-horicromtal.yml:9]: TOL I'd probably make these opt-in rather than opt-out

**Pull request merged**